service-csw/src/main/java/org/oskari/csw/request/GetRecords.java

```diff
@@ -0,0 +1,177 @@
+package org.oskari.csw.request;
+
+import fi.nls.oskari.service.ServiceRuntimeException;
+import org.geotools.filter.v1_0.OGCConfiguration;
+import org.geotools.xml.Encoder;
+import org.opengis.filter.Filter;
+
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.Charset;
+
+public class GetRecords {
+
+    private static final XMLOutputFactory xof = XMLOutputFactory.newFactory();
+    private static final String CSW_URI = "http://www.opengis.net/cat/csw/2.0.2";
+    private static final String GMD_URI = "http://www.isotc211.org/2005/gmd";
+
+    private static final String CSW_VERSION = "2.0.2";
+    private static final String CONSTRAINT_VERSION = "1.0.0";
+
+    private GetRecords() {}
+
+    /**
+     * Builds a GetRecords request payload for CSW-service with the given filters.
+     * @param filter required
+     * @return
+     */
+    public static String createRequest(Filter filter) {
+        if (filter == null) {
+            throw new ServiceRuntimeException("Filter is required");
+        }
+        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        XMLStreamWriter xsw = getWriter(stream);
+        startDocument(xsw);
+        writeRawXMLUnsafe(xsw, stream, getFilterAsString(filter));
+        endDocument(xsw);
+        return getAsString(stream);
+    }
+
+    /**
+     * Nothing fancy. Just wrapping possible exception to a runtime exception.
+     * @param stream
+     * @return
+     */
+    private static XMLStreamWriter getWriter(OutputStream stream) {
+        try {
+            return xof.createXMLStreamWriter(stream);
```

Comment:
Missing explicit charset information. Can pass a Writer instead of an OutputStream here.